app/src/main/java/org/android/go/sopt/util/ContextUtil.kt

```diff
@@ -23,4 +27,25 @@ fun Context.hideKeyboard(view: View) {
 
 fun Context.showToast(msg: String) {
     Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
+}
+
+fun <ResponseType> Call<ResponseType>.enqueueUtil(
+    onSuccess: (ResponseType) -> Unit,
+    onError: ((stateCode: Int) -> Unit)? = null
+) {
+    this.enqueue(object : Callback<ResponseType> {
+        override fun onResponse(call: Call<ResponseType>, response: Response<ResponseType>) {
+            if (response.isSuccessful) {
+                onSuccess.invoke(response.body() ?: return)
+            } else {
+                onError?.invoke(response.code())
+            }
+        }
```

Comment:
```suggestion
    onError: ((stateCode: Int, message: String) -> Unit)? = null
) {
    this.enqueue(object : Callback<ResponseType> {
        override fun onResponse(call: Call<ResponseType>, response: Response<ResponseType>) {
            if (response.isSuccessful) {
                onSuccess.invoke(response.body() ?: return)
            } else {
                onError?.invoke(response.code(), response.message())
            }
        }
```
생각해보니 Response 클래스 안에는 code()와 message()라는 메서드가 있습니다. 각각 statusCode(Int)와 message(String)를 반환하는 메서드입니다. 따라서 when 문으로 특정 code에 대해 String을 직접 Toast를 통해 보여주지 않고 바로 message를 전달해 Toast Message를 보여줄 수 있을 것 같아요.